lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -0,0 +1,162 @@
+    txMetaSlotId     W.SlotId     sql=slot_id
+    txMetaAmount     W.Coin       sql=amount
+
+    Primary txId txMetaWalletId
+    Foreign Wallet fk_wallet_tx_meta txMetaWalletId
+    deriving Show Generic
+
+-- A transaction input associated with TxMeta.
+--
+-- There is no wallet ID because these values depend only on the transaction,
+-- not the wallet. txInputTxId is referred to by TxMeta and PendingTx
+TxIn
+    txInputTxId         TxId    sql=tx_id
+    txInputSourceTxId   TxId    sql=source_id
+    txInputSourceIndex  Word32  sql=source_index
+    txInputAddress      Text    sql=address
+    txInputAmount       W.Coin  sql=amount
+
+    Primary txInputTxId txInputSourceTxId txInputSourceIndex
+    -- fixme: add index on tx_id
+    deriving Show Generic
+
+-- A transaction output associated with TxMeta.
+--
+-- There is no wallet ID because these values depend only on the transaction,
+-- not the wallet. txOutputTxId is referred to by TxMeta and PendingTx
+TxOut
+    txOutputTxId     TxId    sql=tx_id
+    txOutputIndex    Word32  sql=index
+    txOutputAddress  Text    sql=address
+    txOutputAmount   W.Coin  sql=amount
+
+    Primary txOutputTxId txOutputIndex
+    -- fixme: add index on tx_id
+    deriving Show Generic
+
+-- A checkpoint for a given wallet is referred to by (wallet_id, slot_id).
+-- Checkpoint data such as UTxO will refer to this table.
+Checkpoint
+    checkpointWalId    W.WalletId  sql=wallet_id
+    checkpointWalSlot  W.SlotId    sql=slot_id
+
+    Primary checkpointWalId checkpointWalSlot
+    Foreign Wallet fk_wallet_checkpoint checkpointWalId
+
+    deriving Show Generic
+
+-- The UTxO for a given wallet checkpoint is a one-to-one mapping from TxIn ->
+-- TxOut. This table does not need to refer to the TxIn or TxOut tables. All
+-- necessary information for the UTxO is in this table.
+UTxO                             sql=utxo
+
+    -- The wallet checkpoint (wallet_id, slot_id)
+    utxoWalletId     W.WalletId  sql=wallet_id
+    utxoWalletSlot   W.SlotId    sql=slot_id
+
+    -- TxIn
+    utxoInputId      TxId        sql=input_tx_id
+    utxoInputIndex   Word32      sql=input_index
+
+    -- TxOut
+    utxoOutputId     TxId        sql=output_tx_id
+    utxoOutputIndex  Word32      sql=output_index
+
+    Primary
+        utxoWalletId
+        utxoWalletSlot
+        utxoInputId
+        utxoInputIndex
+        utxoOutputId
+        utxoOutputIndex
+
+    Foreign Checkpoint fk_checkpoint_utxo utxoWalletId utxoWalletSlot
+
+    deriving Show Generic
+
+-- The pending transactions for a wallet checkpoint.
+PendingTx
+
+    -- The wallet checkpoint (wallet_id, slot_id)
+    pendingTxWalletId    W.WalletId  sql=wallet_id
+    pendingTxSlotId      W.SlotId    sql=slot_id
+
+    -- Transaction TxIn and TxOut
+    pendingTxId2         TxId        sql=tx_id
+
+    Primary pendingTxWalletId pendingTxSlotId pendingTxId2
+    Foreign Checkpoint fk_pending_tx pendingTxWalletId pendingTxSlotId
+
+    deriving Show Generic
```

Comment:
What's the need for this since we already have a transaction table?

Reply:
removed PendingTx

Reply:
I think it's needed to associate pending transactions with a checkpoint. But we'll find out later when implementing `putCheckpoint` and `readCheckpoint`.